scheds/c/scx_central.c

```diff
@@ -72,6 +77,11 @@ int main(int argc, char **argv)
 			u32 central_cpu = strtoul(optarg, NULL, 0);
 			if (central_cpu >= skel->rodata->nr_cpu_ids) {
 				fprintf(stderr, "invalid central CPU id value, %u given (%u max)\n", central_cpu, skel->rodata->nr_cpu_ids);
+				free(cpumask);
+				return -1;
+			} else if (central_cpu >= ncpu || !cpumask[central_cpu]) {
```

Comment:
I'm not sure how valuable it is to add just CPU online test here. Overall, the logic is still wrong as it's still assuming possible cpumasks don't have gaps.

Reply:
So what would be your suggestion ?

Reply:
As it's more of a simple example to demonstrate an idea, I wonder whether we should just leave it be. We can add possible / online distinction and hotplug handling but that seems rather too much given that the scheduler isn't a very practical one to begin with. We have similar issues on rust scheds and there we should do it properly but it'd also be a lot easier to do so as it's easy to grow libraries and helpers there.